src/SFA.DAS.EAS.Portal/Application/Services/Commitments/Http/CommitmentsApiHttpClientFactory.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Net.Http;
+using SFA.DAS.EAS.Portal.Configuration;
+using SFA.DAS.Http;
+using SFA.DAS.Http.TokenGenerators;
+
+namespace SFA.DAS.EAS.Portal.Application.Services.Commitments.Http
+{
+    public class CommitmentsApiHttpClientFactory : ICommitmentsApiHttpClientFactory
+    {
+        private readonly CommitmentsApiClientConfiguration _commitmentsApiClientConfiguration;
+
+        public CommitmentsApiHttpClientFactory(CommitmentsApiClientConfiguration commitmentsApiClientConfiguration)
+        {
+            _commitmentsApiClientConfiguration = commitmentsApiClientConfiguration;
+        }
+
+        public HttpClient CreateHttpClient()
+        {
+            var httpClient = new HttpClientBuilder()
+                .WithDefaultHeaders()
+                .WithBearerAuthorisationHeader(new JwtBearerTokenGenerator(_commitmentsApiClientConfiguration))
+                .Build();
+
+            httpClient.BaseAddress = new Uri(_commitmentsApiClientConfiguration.BaseUrl);
+
+            //todo: set timeout
+            //httpClient.Timeout = TimeSpan.Parse(_recruitApiClientConfig.TimeoutTimeSpan);
```

Comment:
if this isn't being done now do we need to make sure there is some tech debt or a story somewhere to address this later down the line ?

Reply:
It's not critical to have a timeout here, as it is when the client call API's, as the call is made in the context of the worker. I'll remove the comment for now, and we can always add a timeout if we really need it.